src/client/components/Editors/Chart/Layout.tsx

```diff
@@ -1,70 +1,69 @@
 import * as React from 'react'
-import camelCase from 'lodash/camelCase'
 import Box from '@mui/material/Box'
 import Columns from '../../Parts/Grids/Columns'
-import VerticalTabs from '../../Parts/Tabs/Vertical'
 import EditorHelp from '../Base/Help'
-import Chart from './Sections/Chart/Chart'
 import Channel from './Sections/Channel/Channel'
 import Transform from './Sections/Transform/Transform'
-import HorizontalTabs from '../../Parts/Tabs/DynamicHorizontal'
 import { useStore } from './store'
+import MenuTree from '../../Parts/Trees/Menu'
+import Chart from './Sections/Chart/Chart'
 import LayerChart from './Sections/Chart/LayerChart'
 
-const LABELS = ['Chart', 'Channels', 'Transforms']
-
 export default function Layout() {
+  const externalMenu = useStore((state) => state.externalMenu)
   return (
     <Box sx={{ height: '100%' }}>
-      <LayoutWithMenu />
+      {!externalMenu ? <LayoutWithMenu /> : <LayoutWithoutMenu />}
     </Box>
   )
 }
 
 function LayoutWithMenu() {
-  const tabIndex = useStore((state) => state.tabIndex)
+  const section = useStore((state) => state.section)
+  const updateHelp = useStore((state) => state.updateHelp)
   const updateState = useStore((state) => state.updateState)
-  const labels = useStore((state) => state.tabNames)
-  const addTab = useStore((state) => state.addLayer)
-  const removeTab = useStore((state) => state.removeLayer)
+  const layerItems = useStore((state) => state.layerItems)
+  const addLayer = useStore((state) => state.addLayer)
+  const layers = useStore((state) => state.layers)
   return (
-    <HorizontalTabs
-      index={tabIndex}
-      labels={labels}
-      isDynamic={true}
-      onChange={(index) => {
-        updateState({ tabIndex: index })
-      }}
-      onAdd={addTab}
-      onRemove={removeTab}
-    >
-      {labels.map((_, index) => {
-        return <LayoutWithoutMenu key={index} />
-      })}
-    </HorizontalTabs>
+    <Columns spacing={3} layout={[2, 8]} columns={10}>
+      <Box sx={{ padding: 2, borderRight: 'solid 1px #ddd', height: '100%' }}>
+        <MenuTree
+          menuItems={layerItems}
+          selected={section}
+          defaultExpanded={['chart']}
+          onAddNew={addLayer}
+          onSelect={(section) => {
+            updateHelp(section)
+            updateState({ section })
+            const layerIndex = layers.findIndex((elem) => elem === section.split('/')[0])
+            updateState({ layerIndex })
+          }}
+        />
+      </Box>
+      <LayoutWithoutMenu />
+    </Columns>
   )
 }
 
 function LayoutWithoutMenu() {
+  const section = useStore((state) => state.externalMenu?.section || state.section)
```

Comment:
I would split state `section` (Chart, Channels, Transforms) and `layer` -- at least not to break the help system